salt/modules/git.py

```diff
@@ -1297,6 +1297,145 @@ def describe(cwd, rev='HEAD', user=None, ignore_retcode=False):
                     ignore_retcode=ignore_retcode)['stdout']
 
 
+def diff(cwd,
```

Comment:
This is really the only function I need but to get a clean history I pulled in a bunch of other commits.